hadoop-hdds/tools/src/main/java/org/apache/hadoop/hdds/scm/cli/ContainerOperationClient.java

```diff
@@ -382,7 +382,7 @@ public ContainerDataProto readContainer(long containerID,
       Pipeline pipeline) throws IOException {
     XceiverClientSpi client = null;
     try {
-      client = xceiverClientManager.acquireClient(pipeline);
+      client = xceiverClientManager.acquireClientForReadData(pipeline);
```

Comment:
I have not really understood this change, as we just changed from acquireClient  -> acquireClientForReadData.
How that will handle failures, and how it will retry on other datanodes.

And also InfoSubCommand Uses below code, and it is not calling readContainer.
      final ContainerWithPipeline container = scmClient.
          getContainerWithPipeline(containerID);

>       final ContainerWithPipeline container = scmClient.
>           getContainerWithPipeline(containerID);
>       Preconditions.checkNotNull(container, "Container cannot be null");

Reply:
@bharat, if we use acquireClientForReadData, it will use XceiverClientGrpc to read the info rather than Ratis(which reads from the leader). XceiverClientGrpc automatically fails over from datanode to another if it sees a failure.

InfoSubCommand has recently been changed to not use readContainer call to read data from datanode. But , in Ozone, we only use XceiverClientGrpc to read from datanode , it make sense here to use the same while reading the containerData as well (acquireClient -> acquireClientForReadData). This function is used by ChunkKeyHandler.